doc/content/getting-started/migrating/migrating-from-v2/migrate-using-migration-tool/establish-new-session.md

```diff
@@ -58,14 +78,32 @@ $ ttn-lw-migrate application --source ttnv2 "ttn-v2-application-ID" --ttnv2.with
 
 Exporting end devices without their active sessions does not clear root and session keys on {{% ttnv2 %}}. Hence, you need to prevent your devices from re-joining the {{% ttnv2 %}} network.
 
+{{< tabs/container "OTAA" "ABP" >}}
+
+{{< tabs/tab "OTAA" >}}
+
 To prevent OTAA device from re-joining the {{% ttnv2 %}} network, the recommended practice is to change the **AppKey** in the {{% ttnv2 %}}. By changing the **AppKey**, the existing session on {{% ttnv2 %}} will not be terminated yet, but the end device will not be able to re-join because the {{% ttnv2 %}} cluster will reject its new Join Requests. 
 
+{{< /tabs/tab >}}
+
+{{< tabs/tab "ABP" >}}
+
+For ABP device, you need to completely delete it from V2, especially because this section assumes that you have not re-programmed it for a new **DevAddr**. Having this device registered in V2 and The Things Stack would introduce some serious conflicts. 
```

Comment:
This part is important right? Let's add the warning tag

```suggestion
{{< warning >}} For ABP device, you need to completely delete it from V2, especially because this section assumes that you have not re-programmed it for a new **DevAddr**. Having this device registered in V2 and The Things Stack would introduce some serious conflicts. {{</ warning >}}
```